network/address_test.go

```diff
@@ -51,12 +53,13 @@ func TestAddress(t *testing.T) {
 	}
 
 	for i, str := range tests {
+		log.Lvl1("Testing", str)
 		add := Address(str.Value)
-		assert.Equal(t, str.Valid, add.Valid(), "Address (%d) %s", i, str.Value)
-		assert.Equal(t, str.Type, add.ConnType(), "Address (%d) %s", i, str.Value)
-		assert.Equal(t, str.Address, add.NetworkAddress())
-		assert.Equal(t, str.Host, add.Host())
-		assert.Equal(t, str.Port, add.Port())
-		assert.Equal(t, str.Public, add.Public())
+		require.Equal(t, str.Valid, add.Valid(), "Address (%d) %s", i, str.Value)
+		require.Equal(t, str.Type, add.ConnType(), "Address (%d) %s", i, str.Value)
+		require.Equal(t, str.Address, add.NetworkAddress())
+		require.Equal(t, str.Host, add.Host())
+		require.Equal(t, str.Port, add.Port())
+		require.Equal(t, str.Public, add.Public())
```

Comment:
Ok I don't mind so much but here is a great example where `assert` is exactly what we need because the things that are tested after a failing assertion are independant from the failing assertion. In short, we can see all errors at once instead of failing over the first one.